src/control/server/mgmt_svc.go

```diff
@@ -220,6 +222,14 @@ func (svc *mgmtSvc) GetAttachInfo(ctx context.Context, req *mgmtpb.GetAttachInfo
 		return nil, errors.Wrap(err, "unmarshal GetAttachInfo response")
 	}
 
+	if svc.agentCfg == nil {
+		return nil, errors.New("agentCfg in GetAttachInfo response was empty")
```

Comment:
This error message is pretty misleading. The agent cfg didn't come in the response, even though it will be added to the response. Would be better to say we don't have an agent config, if we strictly need one, before we bother with calling dRPC.

Reply:
I've moved the check earlier and changed the error text to "agentCfg is missing".